test/acceptance/backup-service-info.test.js

```diff
@@ -0,0 +1,123 @@
+/*global describe, it, before, after */
+
+const backup = require('../../src/main');
+const credentials = require('../helpers/testuser').credentialsServiceInfo;
+const async = require('async');
+const fs = require('fs');
+const should = require('should');
+const pryv = require('pryv');
+const parseDomain = require("parse-domain");
+
+describe('backup with service info', function () {
+
+  this.timeout(10000);
+
+  let settings = null;
+  let resources = null;
+  let connection = null;
+
+  before(function (done) {
+    settings = {
+      username: credentials.username,
+      domain: credentials.domain,
+      password: credentials.password,
+      includeTrashed: true,
+      includeAttachments: true,
+      appId: 'pryv-backup'
+    };
+
+    settings.origin = 'https://sw.' + settings.domain;
+    settings.backupDirectory = new backup.Directory(settings.username, settings.domain);
+    // console.log('settings : '+JSON.stringify(settings, null, 2));
+
+    const eventsRequest = 'events?fromTime=-2350373077&toTime=' + new Date() / 1000 + '&state=all';
+    const streamsRequest = 'streams?state=all';
+    resources = ['account', streamsRequest, 'accesses', 'followed-slices', 'profile/public', eventsRequest];
+    
+    const parsedDomain = parseDomain(settings.domain);
+    settings.domain = parsedDomain.domain + '.' + parsedDomain.tld;
+
+    pryv.Connection.login(settings, function (err, conn) {
+      connection = conn;
+      settings.backupDirectory.deleteDirs(done);
+    });
+  });
+
+  after(function (done) {
+    settings.backupDirectory.deleteDirs(done);
+  });
+
+  it('should backup the correct folders and files', function (done) {
+    const time = Date.now()/1000;
+    async.series([
```

Comment:
Pour l'amour de dieu arrêtons d'ajouter du code avec callbacks et async.series/each.
Plus sérieusement, comme évoqué plus haut je pense qu'on gagne en lisibilité si on écrit ce genre de structure avec async/await (et bluebird.fromCallback pour les fonctions qui ne suivent pas encore ce schema).

En parlant de lisibilité, il y a une structure qui marche assez bien pour les tests, tu pourrais par exemple start le backup dans un before, puis avoir plusieurs block 'it' qui chacun test une chose (ex: it ('backup the files'), it('backup the attachments')...)

Reply:
Alors là je t'avoue que par pur flemme j'ai juste fait un vieux copier/coller de backup.test.js !